components/virtual-node/virtual-node-write-service-impl/src/main/kotlin/net/corda/virtualnode/write/db/impl/writer/VirtualNodeWriterProcessor.kt

```diff
@@ -79,64 +81,129 @@ internal class VirtualNodeWriterProcessor(
         const val PUBLICATION_TIMEOUT_SECONDS = 30L
     }
 
-    @Suppress("ReturnCount")
+    @Suppress("ReturnCount", "ComplexMethod")
     private fun createVirtualNode(
         instant: Instant,
         create: VirtualNodeCreateRequest,
         respFuture: CompletableFuture<VirtualNodeManagementResponse>
     ) {
-        create.validationError()?.let { errMsg ->
-            handleException(respFuture, IllegalArgumentException(errMsg))
-            return
+        // TODO - replace this with real metrics
+        logger.info("Create new Virtual Node: ${create.x500Name} and ${create.cpiFileChecksum}")
+        val startMillis = currentTimeMillis()
+
+        measureTimeMillis {
+            create.validationError()?.let { errMsg ->
+                handleException(respFuture, IllegalArgumentException(errMsg))
+                return
+            }
+        }.also {
+            logger.debug("[Create ${create.x500Name}] validation took $it ms, elapsed " +
+                    "${currentTimeMillis() - startMillis} ms")
```

Comment:
Let's use labda version for `debug`: 
```suggestion
            logger.debug { "[Create ${create.x500Name}] validation took $it ms, elapsed " +
                    "${currentTimeMillis() - startMillis} ms" }
```

Reply:
There is also a utility: `net.corda.utilities.LogUtils#logElapsedTime` but it logs at `info` level all the time currently.

Reply:
Y. I don't think logging metrics at info level is right. 
The problem with untils like that is that they don't abstract too much and introduce limitations. For this, I thought it was quite important to capture individual times as well as the elapsed. Regardless of what Detekt says, I don't think it's made the code particularly less readable.
Off course, we should be exposing these as metrics not really logs.